README.md

```diff
@@ -48,3 +48,151 @@ To test with Nextflow for development purpose:
 
 * Java 11 or later
 * https://github.com/everit-org/json-schema
+
+# Usage
+
+## Introduction
+
+The nf-validation plugin implements the validation of Nextflow pipeline parameters using the [JSON Schema](https://json-schema.org/) format. This plugin contains different functions that can be included into a Nextflow pipeline to provide documentation or validate the pipeline parameters.
+
+## Quick Start
+
+Declare the plugin in the Nextflow configuration file:
+
+```
+plugins {
+  id 'nf-validation@0.0.1'
+}
+```
+
+Include a function into your Nextflow pipeline and execute it.
+
+```
+include { validateParameters } from 'plugin/nf-validation'
+validateParameters()
+```
+
+You can find more information on plugins in the [Nextflow documentation](https://www.nextflow.io/docs/latest/plugins.html#plugins).
+
+### The JSON Schema
+
+A JSON schema file will contains the information for all pipeline parameters. It can define the required parameters, describe parameter characteristics such as type (ex. string, integer), the pattern (regular expressions matching strings) or a description of the parameter.
+
+You can find more information about JSON Schemas in their official [documentation webpage](https://json-schema.org/). You can see an example JSON Schema for a Nextflow pipeline [nextflow_schema.json](https://raw.githubusercontent.com/nextflow-io/nf-validation/master/plugins/nf-validation/src/testResources/nextflow_schema.json).
```

Comment:
We should add a note that we don't support the full schema spec. For example, no arrays or objects, and we have a whole bunch of custom keys.

My preferred way to do this is to actually write proper docs on the nf-core website and refer to them from here.

Reply:
I added a note and the link to the nf-core documentation which explains this, but we could make it more obvious on the nf-core website.

Reply:
Definitely agreed. I've been meaning to write these for.. years now!